### PR TITLE
Can HW acceptance filter feature + getIface() corrections + dsdl hash fixed

### PR DESCRIPTION
**Pull request by ilia-sheremet**

### CODE REVIEW

File: libuavcan/src/transport/uc_can_acceptance_filter_configurator.cpp [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+    while (p)
+    {
+        CanFilterConfig cfg;
+        cfg.id = static_cast<uint32_t>(p->getDataTypeDescriptor().getID().get()) << 16;
+        cfg.id |= static_cast<uint32_t>(p->getDataTypeDescriptor().getKind()) << 8;
+        cfg.mask = DefaultFilterMsgMask;
+        if (multiset_configs_.emplace(cfg) == NULL)
+            {
+                return -ErrMemory;
+            }
```
Comment (pavel-kirienko) [inline libuavcan/src/transport/uc_can_acceptance_filter_configurator.cpp]: Formatting.

File: libuavcan/src/transport/uc_can_acceptance_filter_configurator.cpp [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+const unsigned CanAcceptanceFilterConfigurator::ServiceRespFrameMask;
+
+int16_t CanAcceptanceFilterConfigurator::loadInputConfiguration()
+{
+    multiset_configs_.clear();
+
+    CanFilterConfig service_resp_cfg;
+    service_resp_cfg.id = ServiceRespFrameID;
+    service_resp_cfg.mask = ServiceRespFrameMask;
+	if (multiset_configs_.emplace(service_resp_cfg) == NULL)
```
Comment (pavel-kirienko) [inline libuavcan/src/transport/uc_can_acceptance_filter_configurator.cpp]: Formatting.

File: libuavcan/src/transport/uc_can_acceptance_filter_configurator.cpp [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+                     multiset_configs_.getByIndex(i)->mask);
+    }
+#endif
+
+    return 0;
+}
+
+int16_t CanAcceptanceFilterConfigurator::computeConfiguration()
+{
+    const uint16_t acceptance_filters_number = getNumFilters();
```
Comment (pavel-kirienko) [inline libuavcan/src/transport/uc_can_acceptance_filter_configurator.cpp]: If `getNumFilters()` returns zero, the algorithm will fail. This corner case should be checked before the loop starts.

File: libuavcan/src/transport/uc_can_acceptance_filter_configurator.cpp [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+            return -ErrDriver;
+        }
+    }
+
+    return 0;
+}
+
+int CanAcceptanceFilterConfigurator::configureFilters()
+{
+    if (getNumFilters() == 0)
```
Comment (pavel-kirienko) [inline libuavcan/src/transport/uc_can_acceptance_filter_configurator.cpp]: This check is nice, but it doesn't protect against transient failures, so `computeConfiguration()` should check it too.

Comment (pavel-kirienko) [inline libuavcan/src/transport/uc_can_acceptance_filter_configurator.cpp]: Alternatively, you could store the number of filters there, and pass the value into `computeConfiguration()`.

**Comment by pavel-kirienko** [comment]
Спасибо.

**Comment by ilia-sheremet** [comment]
Всегда пожалуйста =)

**Pull request merged**